src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -575,5 +575,54 @@ private static IEnumerable<PropertyInfo> GetAllProperties(Type type)
 
             return allProperties;
         }
+
+        private static object GetPropertyValue(PropertyInfo property, object instance, IConfiguration config, BinderOptions options, object defaultValue)
+        {
+            try
+            {
+                string propertyName = GetPropertyName(property);
+                return BindInstance(
+                    property.PropertyType,
+                    property.GetValue(instance),
+                    config.GetSection(propertyName),
+                    options) ?? defaultValue;
+            }
+            catch
```

Comment:
Not a big fan of a try catch like this, especially if we didn't catch all exceptions before

Reply:
Running tests at the test project was passing locally. Swapped over to test others and found that it was breaking expected exception behavior. Removed and pushed.

Reply:
This method should handle the error cases explicitly only, I still posit that property should never be null, maybe instance can be null, but something has gone really wrong if null gets into GetPropertyName, so you don't want to swallow that kind of exception.  I don't believe we check for property to be non null elseware, so you could just drop the check and let it null ref too.